demos/swipe_detection/swipe_detection_network.py

```diff
@@ -0,0 +1,266 @@
+from lava.proc.embedded_io.spike import NxToPyAdapter, PyToNxAdapter
+from lava.utils.serialization import save
+from outprocess.process_out import ProcessOut
+from scipy.sparse import csr_matrix
+from lava.magma.compiler.subcompilers.nc.ncproc_compiler import CompilerOptions
+
+CompilerOptions.verbose = True
+
+prophesee_input_default_config = {
+    "filename": "hand_swipe.dat",
+    "transformations": Compose(
+        [
+            Downsample(factor=0.125),
+            MergePolarities(),
+        ]
+    ),
+    "sensor_shape": (480, 640),
+    "num_output_time_bins": 1,
+    "sync_time": False,
+    "flatten": True
+}
+
+lif_default_config = {
+    "vth": 600,
+    "du": 2000,
+    "dv": 2000
+}
+
+weights_default_config = {
+    "w_ff": 255,
+    "w_rec": 5,
+    "w_o": 16,
+    "kernel_size": 20
+}
+
+num_out_neurons_default = 50
+blocking_default_config = False
+
+
+class SwipeDetector:
+    """Class to setup the swipe detection network, compile it, and initialize
+         its runtime."""
+
+    def __init__(self,
+                 send_pipe: type(Pipe),
+                 num_steps: int,
+                 use_loihi2: bool,
+                 prophesee_input_config: ty.Optional[dict] = None,
+                 lif_config: ty.Optional[dict] = None,
+                 weights_config: ty.Optional[dict] = None,
+                 num_out_neurons: ty.Optional[dict] = None,
+                 blocking: ty.Optional[bool] = None,
+                 executable: ty.Optional[Executable] = None
+                 ) -> None:
+        self.prophesee_input_config = prophesee_input_config or \
+            prophesee_input_default_config
+        self.lif_config = lif_config or lif_default_config
+        self.weights_config = weights_config or weights_default_config
+        self.num_out_neurons = num_out_neurons or num_out_neurons_default
+        self.blocking = blocking or blocking_default_config
+        self.send_pipe = send_pipe
+        self.num_steps = num_steps
+        self.use_loihi2 = use_loihi2
+        self._create_processes()
+        self._make_connections()
+        print("network created")
+
+        # Run
+        if self.use_loihi2:
+            print("running on chip")
+            run_cfg = Loihi2HwCfg(exception_proc_model_map=
+                                  {PropheseeCamera: PyPropheseeCameraRawReaderModel})
+        else:
+            run_cfg = Loihi2SimCfg(
+                exception_proc_model_map={Dense: PyDenseModelBitAcc,
+                                          Sparse: PySparseModelBitAcc,
+                                          LIF: PyLifModelBitAcc,
+                                          PropheseeCamera: PyPropheseeCameraRawReaderModel})
+
+        # Compilation
+        if executable is None:
+            compiler = Compiler()
+            executable = compiler.compile(self.frame_input, run_cfg=run_cfg)
+        # Store the Lava network, only needed if network changes
+        #save(processes=[self.ff_inp,
```

Comment:
Could this code live in a method and be triggered using a parameter given at the instantiation of the class?
This parameter could for example be a path/filename to store the network in. If given, the network is stored, if not given, then not.

Reply:
Done